datafusion/core/Cargo.toml

```diff
@@ -127,6 +127,7 @@ parquet = { workspace = true, optional = true, default-features = true }
 paste = "1.0.15"
 pin-project-lite = "^0.2.7"
 rand = { workspace = true }
+regex = { workspace = true }
```

Comment:
I believe this is not a new actual dependency as it is already a dependency of `arrow-csv`

Reply:
When I remove it I have this error during `cargo build`:

```
error[E0432]: unresolved import `regex`
  --> datafusion/core/src/datasource/file_format/csv.rs:59:5
   |
59 | use regex::Regex;
   |     ^^^^^ help: a similar path exists: `datafusion_functions::regex`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `datafusion` (lib) due to 1 previous error
```

Reply:
Yeah, the "better" think to do would be for arrow-rs to re-export the RegEx structure it used as it appears in the public API